airbyte/caches/ducklake.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+
+"""A DuckLake implementation of the PyAirbyte cache, built on DuckDB.
+
+```python
+from airbyte as ab
+from airbyte.caches import DuckLakeCache
+
+cache = DuckLakeCache(
+    metadata_connection_string="sqlite:./metadata.db",
+    data_path="./ducklake_data/",
+    catalog_name="my_catalog",
+    schema_name="myschema",
+)
+```
+"""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, ClassVar
+
+from overrides import overrides
+from pydantic import Field
+
+from airbyte._processors.sql.duckdb import DuckDBConfig, DuckDBSqlProcessor
+from airbyte.caches.duckdb import DuckDBCache
+from airbyte.secrets import SecretString
+
+
+if TYPE_CHECKING:
+    from airbyte.shared.sql_processor import SqlProcessorBase
+
+
+class DuckLakeConfig(DuckDBConfig):
+    """Configuration for the DuckLake cache."""
+
+    db_path: Path | str = Field(default="ducklake.db")
```

Comment:
This should use the 'duckdb' extension and have a more explicit db name, to make it clear. I will commit the folllowing and you can pull to get the latest.

```suggestion
    db_path: Path | str = Field(default="ducklake-dummy-db.duckdb")
```